roxie/ccd/ccdfile.cpp

```diff
@@ -1578,69 +1588,88 @@ class CRoxieFileCache : implements IRoxieFileCache, implements ICopyFileProgress
             DBGLOG("HandleCloser thread %p starting", this);
         try
         {
-            unsigned lastCloseCheck = msTick();
-            for (;;)
-            {
+            unsigned now = msTick();
 #ifdef _CONTAINERIZED
-                unsigned checkPeriod = topology->getPropInt("@copyCheckPeriod", 60);
-#else
-                unsigned checkPeriod = 10*60;  // check expired file handles every 10 minutes, buddyCopying a little more often
+            unsigned lastBuddyCheck = now;
 #endif
-                bool forceElapsedCheck = toClose.wait(checkPeriod * 1000);
+            unsigned lastCloseCheck = now;
+            unsigned lastReopenReport = now;
+            unsigned lastFilesReopened = filesReopened;
+            unsigned __int64 lastReopenedDelay = reopenedDelay;
+            for (;;)
+            {
+                bool forceElapsedCheck = toClose.wait(60 * 1000);
                 if (closing)
                     break;
-#ifdef _CONTAINERIZED
-                // Periodically recheck the list to see what is now local, and remove them from the buddyCopying list
-                IArrayOf<ILazyFileIO> checkBuddies;
+                now = msTick();
+                if (now - lastReopenReport >= 60*1000)
                 {
-                    CriticalBlock b(crit);
-                    while (buddyCopying.length())
+                    unsigned newReopened = filesReopened - lastFilesReopened;
+                    if (newReopened)
                     {
-                        ILazyFileIO *popped = &buddyCopying.popGet();
-                        if (popped->isAliveAndLink())
-                        {
-                            checkBuddies.append(*popped);
-                        }
+                        DBGLOG("%u files reopened in last %u seconds (average closed time %" I64F "u ms)", 
+                               newReopened, (now - lastReopenReport)/1000, 
+                               nanoToMilli((reopenedDelay - lastReopenedDelay)/newReopened));
+                        lastFilesReopened = filesReopened;
+                        lastReopenedDelay = reopenedDelay;
                     }
-                    buddyChecking = true;
+                    lastReopenReport = now;
                 }
-                if (checkBuddies.length())
+#ifdef _CONTAINERIZED
+                if (now - lastBuddyCheck >= 60*1000)
```

Comment:
lost configuration for the check period.  Does that matter?
unsigned copyCheckPeriod = topology->getPropInt("@copyCheckPeriod", 60);?
...
?

Reply:
It was intentional, to simplify the code. The config option was never documented and was really only there to assist in testing the functionality.